reporters/dogstatsd.go

```diff
@@ -84,3 +102,23 @@ func NewDogStatsD(config *viper.Viper, logger *logrus.Logger) (*DogStatsD, error
 func NewDogStatsDFromClient(c dogstatsd.Client, r string) *DogStatsD {
 	return &DogStatsD{client: c, region: r}
 }
+
+func (d *DogStatsD) scheduleRestart(timeout time.Duration) {
+	if timeout.Nanoseconds() > 0 {
+		time.AfterFunc(timeout, func() {
+			if err := d.statsdClient.Close(); err != nil {
+				d.logger.Errorf("DogStatsD: failed to close statsd connection during restart: %s", err.Error())
+			}
+			
+			c, err := dogstatsd.New(d.host, d.prefix)
+			if err == nil {
+				d.statsdClient = c.Client.(*statsd.Client)
```

Comment:
This isn't thread-safe, right? The goroutine calling this would start failing. Would we let it fail (since losing a few metrics is fine) or control it through a lock? If we'll let it fail, we need to at least document it through comments and turn any error log into a warning log. WDYT?